File: lib/gtfs/minischedules/trip.ex
defmodule Gtfs.Minischedules.Trip do
  alias Gtfs.Route
  alias Gtfs.Hastus.Place

  @type t :: %__MODULE__{
    route: Route.id() | nil,
    start_time: String.t(),
    end_time: String.t(),
    start_place: Place.id(),
    end_place: Place.id(),
  }

  @enforce_keys [
    :start_time,
    :end_time,
    :start_place,
    :end_place
  ]

  defstruct [
    :route,
    :start_time,
    :end_time,
    :start_place,
    :end_place
  ]
end

File: assets/tests/factories/trip.ts
import { Factory } from "fishery"
import { Trip } from "../../src/minischedule"
import { StopTimeFactory } from "./stopTimeFactory"

export const TripFactory = Factory.define<Trip>(({ sequence }) => {
  const duration = 10
  return {
    id: `trip-${sequence}`,
    blockId: "block-id",
    routeId: "route-id",
    headsign: "headsign-Revenue",
    directionId: 1,
    viaVariant: null,
    runId: "run-id",
    startTime: sequence * duration,
    endTime: (sequence + 1) * duration,
    startPlace: "trip-start-place",
    endPlace: "trip-end-place",
    stopTimes: StopTimeFactory.buildList(3),
  }
})

export const DeadheadTripFactory = TripFactory.params({ routeId: null })
